File: Code_VB/C15_UserControls/CustomControls/BookCounter.vb
Imports System.ComponentModel 
Imports System.Web.UI 
Imports System.Web.UI.WebControls 

Namespace CustomControls 
    <ToolboxData("<{0}:BookCounter runat=server></{0}:BookCounter>")> _ 
    Public Class BookCounter 
        Inherits WebControl 
        Implements INamingContainer 
        ' initialize the counted button member 
        Private btn As New CountedButton2("inquiries") 
        
        Public Property BookName() As String 
            Get 
                Return DirectCast(ViewState("BookName"), String) 
            End Get 
            
            Set(ByVal value As String) 
                ViewState("BookName") = value 
            End Set 
        End Property 
        
        Public Property Count() As Integer 
            Get 
                Return btn.Count 
            End Get 
            Set(ByVal value As Integer) 
                btn.Count = value 
            End Set 
        End Property 
        
        Public Sub Reset() 
            btn.Count = 0 
        End Sub 
        
        Protected Overloads Overrides Sub CreateChildControls() 
            Controls.Add(btn) 
        End Sub 
    End Class 
End Namespace 

File: Code_CS/C15_UserControls/CustomControls/BookCounter.cs
using System.ComponentModel;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace CustomControls
{
   [ToolboxData("<{0}:BookCounter runat=server></{0}:BookCounter>")]
   public class BookCounter : WebControl, INamingContainer
   {
      // intialize the counted button member
      CountedButton2 btn = new CountedButton2("inquiries");

      public string BookName
      {
         get
         {
            return (string)ViewState["BookName"];
         }

         set
         {
            ViewState["BookName"] = value;
         }
      }

      public int Count
      {
         get
         {
            return btn.Count;
         }
         set
         {
            btn.Count = value;
         }
      }

      public void Reset()
      {
         btn.Count = 0;
      }

      protected override void CreateChildControls()
      {
         Controls.Add(btn);
      }
   }
}
